blosc/blosc2.c

```diff
@@ -1018,18 +1018,20 @@ static int blosc_d(
 
   neblock = bsize / nstreams;
   for (int j = 0; j < nstreams; j++) {
-    srcsize -= sizeof(int32_t);
-    if (srcsize < 0) {
+    if (srcsize < sizeof(int32_t)) {
       /* Not enough input to read compressed size */
       return -1;
     }
+    srcsize -= sizeof(int32_t);
     cbytes = sw32_(src);      /* amount of compressed bytes */
-    src += sizeof(int32_t);
-    srcsize -= cbytes;
-    if (srcsize < 0) {
-      /* Not enough input to read compressed bytes */
-      return -1;
+    if (cbytes > 0) {
```

Comment:
Do we need to check for `cbytes <= 0` too?

Reply:
That is handled further down in the code.